### DISCUSSION TITLE
Not starting app if it’s already running

### DISCUSSION THREAD

**Discussion by erikmartinessanches** [ttscoff/bunch-docs#311] [Q&A] [answered]
Hi, interesting application. How do I start Visual Studio Code if it's not already running? I tried this but it doesn’t work.
```
---
title: Frontend
---

// Comments start with a hash symbol (#), a double slash (//)
// or are surrounded by /*...*/. blank lines are ignored

// Use @@ to hide all visible apps. Usually used at the top
// of the file before launching the rest of the bunch
# @@

// Apps to launch start at the beginning of a line, no
// need to quote or escape.

if Visual Studio Code is not running
    %Visual Studio Code

// # %Warp


/* Apps launched by a Bunch are quit when the Bunch is closed
   Add a percent symbol to ignore it when closing the Bunch
   (i.e. keep it running) */
# %Terminal

// Include a line starting with a dash to open specific
// files in the app
%Finder
- /Users/erik/Documents/GitHub/frontend


// Integrating with Moom.
* tell application "Moom" to arrange windows according to snapshot "Frontend" ~5

```

**Comment by ttscoff** [comment]
VS Code has taken some special handling and I may have messed something up in the process. I’ll have to look into it.

What audio feedback are you hearing? Can I see your Bunch?

Thanks,
Brett

**Reply by erikmartinessanches** [reply]
Hi, I've updated the question and posted the bunch in its entirety. (I'm also on M1.)

I was hearing "Example Bunch Launched" when launching the provided `Example - Edit First!` bunch. I don't hear it from starting my bunch above, so it's not an issue anymore.

I now noticed that VS Code’s behaviour wIth simply `%Visual Studio Code` (without if statement) is quite close to what I’m after. If VS Code not running, then it will start (and not close on bunch stop). If VS Code is running, then it will show the window (and not open a new window).

**Comment by ttscoff** [comment, marked as answer]
Looking at your Bunch, you're missing an `end` statement on the `if` block, but I don't know what all that's effecting. And the audio feedback is coming from the `say` command in the Example bunch, which is there as… an example :).

I tested your markup with the end statement added and VSC launched and didn't quit when I closed the bunch. I would note that the `if not running` statement is probably not necessary, if the app is already running the worst that will happen is it will be focused. But try adding the `end` statement and see if you get better results.

```
if Visual Studio Code is not running
	% visual studio code
end
```

**Reply by erikmartinessanches** [reply]
You’re right, I missed the `end`!
